[pytorch] fix labs warning in THTensorMoreMath.cpp

ljk53 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #20000 [pytorch] exclude some caffe2 modules from libtorch mobile build
* #19762 [pytorch] CMakeLists changes to enable libtorch for Android
* #19999 [pytorch] fix labs warning in THVectorDefault.cpp
* **#19760 [pytorch] fix labs warning in THTensorMoreMath.cpp**

Summary:
Seems pretty innocent change copied from https://github.com/pytorch/pytorch/pull/16242

Test Plan:
Make sure the following warning is gone:
../aten/src/TH/generic/THTensorMoreMath.cpp:1140:34: note: use function 'std::abs' instead
LAB_IMPLEMENT_BASIC_FUNCTION(abs,labs)
                                 ^~~~
                                 std::abs

Differential Revision: [D15087655](https://our.internmc.facebook.com/intern/diff/D15087655)

ljk53 committed:
[pytorch] fix labs warning in THTensorMoreMath.cpp

Summary:
Seems pretty innocent change copied from https://github.com/pytorch/pytorch/pull/16242

Test Plan:
Make sure the following warning is gone:
../aten/src/TH/generic/THTensorMoreMath.cpp:1140:34: note: use function 'std::abs' instead
LAB_IMPLEMENT_BASIC_FUNCTION(abs,labs)
                                 ^~~~
                                 std::abs

ljk53 committed:
Update on "[pytorch] fix labs warning in THTensorMoreMath.cpp"

ljk53 committed:
Update on "[pytorch] fix labs warning in THTensorMoreMath.cpp"

[pytorch] fix labs warning in THTensorMoreMath.cpp

Summary:
Seems pretty innocent change copied from https://github.com/pytorch/pytorch/pull/16242

Test Plan:
Make sure the following warning is gone:
../aten/src/TH/generic/THTensorMoreMath.cpp:1140:34: note: use function 'std::abs' instead
LAB_IMPLEMENT_BASIC_FUNCTION(abs,labs)
                                 ^~~~
                                 std::abs

gh-metadata: pytorch pytorch 19760 gh/ljk53/12/head

gchanan reviewed (approved):
can you clarify a few things:

1) does this warning only happen on mobile builds?  I don't see it on CPU builds.
2) is that the entire warning?  Seems strange to start with a `note`.
3) none of the other *abs functions warn in this way?

ljk53 commented:
@gchanan to your questions:
1. I see it when I compile with Android NDK r17/r19 + clang;
2. The entire warning:
```
[661/793] Building CXX object caffe2/CMakeFiles/caffe2.dir/__/aten/src/TH/THTensorMoreMath.cpp.o
In file included from /data/users/liujiakai/src/pytorch/aten/src/TH/THTensorMoreMath.cpp:7:
In file included from ../aten/src/TH/THGenerateAllTypes.h:11:
In file included from ../aten/src/TH/THGenerateIntTypes.h:14:
In file included from TH/generic/THTensorMoreMath.cpp:1:
../aten/src/TH/generic/THTensorMoreMath.cpp:1142:34: warning: absolute value function 'labs' given an argument of type 'int64_t' (aka 'long long') but has parameter of type 'long' which may cause truncation of value [-Wabsolute-value]
LAB_IMPLEMENT_BASIC_FUNCTION(abs,labs)
                                 ^
../aten/src/TH/generic/THTensorMoreMath.cpp:1142:34: note: use function 'std::abs' instead
LAB_IMPLEMENT_BASIC_FUNCTION(abs,labs)
                                 ^~~~
                                 std::abs
```

3. There are other *abs warns, e.g.: TH/generic/THVectorDefault.cpp; I'm not entirely sure why @t-vi modified this specifically. I'm just trying to split it out from his PR. Let me know if you want me to fix others in this PR / separately.

ljk53 committed:
Update on "[pytorch] fix labs warning in THTensorMoreMath.cpp"

[pytorch] fix labs warning in THTensorMoreMath.cpp

Summary:
Seems pretty innocent change copied from https://github.com/pytorch/pytorch/pull/16242

Test Plan:
Make sure the following warning is gone:
../aten/src/TH/generic/THTensorMoreMath.cpp:1140:34: note: use function 'std::abs' instead
LAB_IMPLEMENT_BASIC_FUNCTION(abs,labs)
                                 ^~~~
                                 std::abs

gh-metadata: pytorch pytorch 19760 gh/ljk53/12/head

ljk53 committed:
Update on "[pytorch] fix labs warning in THTensorMoreMath.cpp"

[pytorch] fix labs warning in THTensorMoreMath.cpp

Summary:
Seems pretty innocent change copied from https://github.com/pytorch/pytorch/pull/16242

Test Plan:
Make sure the following warning is gone:
../aten/src/TH/generic/THTensorMoreMath.cpp:1140:34: note: use function 'std::abs' instead
LAB_IMPLEMENT_BASIC_FUNCTION(abs,labs)
                                 ^~~~
                                 std::abs

gh-metadata: pytorch pytorch 19760 gh/ljk53/12/head
